### PR TITLE
More data validity checks in PlotCurveItem

### PR DESCRIPTION
**Pull request by termim**
More data validity checks in PlotCurveItem

### CODE REVIEW

File: pyqtgraph/graphicsItems/PlotCurveItem.py [hunk 1]
```diff
@@ -194,8 +196,10 @@ def pixelPadding(self):
     def boundingRect(self):
         if self._boundingRect is None:
             (xmn, xmx) = self.dataBounds(ax=0)
+            if xmn is None or xmx is None:
+                return QtCore.QRectF()
             (ymn, ymx) = self.dataBounds(ax=1)
-            if xmn is None or ymn is None:
```
Comment (ixjlyons) [inline pyqtgraph/graphicsItems/PlotCurveItem.py:198]: It seems to me the `dataBounds` interface is such that either both members or neither member of the tuple can be `None`, so the original check appears to be sufficient. Have you run into a case where that's not true?

Comment (termim) [reply pyqtgraph/graphicsItems/PlotCurveItem.py:198]: Yes it did happen when I used `LinearRegionItem` to zoom into some data plot. It is possible that this was just user mistake and the GUI should check for that, but it is good to check anyway.
BTW the same checks are performed in `ScatterPlotItem.boundingRect`.
Also `dataBounds` interface is somewhat vague - some items can return None as well.

Comment (ixjlyons) [reply pyqtgraph/graphicsItems/PlotCurveItem.py:198]: Fair enough.

**Comment by ixjlyons** [comment]
Thanks @termim

**Pull request merged**